src/hooks/useCheckout/useCheckout.ts

```diff
@@ -0,0 +1,94 @@
+import { useMutation } from '@tanstack/react-query';
+import {
+  GroupBookingRequestResponse,
+  OrganizerInformation,
+  Quote
+} from '@windingtree/glider-types/dist/win';
+import { useAppDispatch, useAppState } from 'src/store';
+import { BookingInfoType } from 'src/store/types';
+import { getTotalRoomCountReducer } from 'src/utils/offers';
+import { bookGroupRequest } from './api';
+import { getBookingMode } from './helpers';
+
+export const useCheckout = () => {
+  const dispatch = useAppDispatch();
+  const { organizerInfo, bookingInfo } = useAppState();
+
+  /**
+   * Save the organizer info in global state
+   * @param info : info of the organizer
+   */
+  const setOrganizerInfo = (info: OrganizerInformation) => {
+    dispatch({
+      type: 'SET_ORGANIZER_INFO',
+      payload: info
+    });
+  };
+
+  /**
+   * Save the booking info in global state
+   * @param info : info of a booking
+   * @param cleanPrevState : overwrite existing state with new state (TRUE), or spread the new state over the existing storage (FALSE).
+   */
+  const setBookingInfo = (info: BookingInfoType, cleanPrevState = false) => {
+    const roomCount = info?.offers?.reduce(getTotalRoomCountReducer, 0);
+
+    const newStore = {
+      ...info,
+      ...(roomCount && { roomCount })
+    };
+
+    dispatch({
+      type: 'SET_BOOKING_INFO',
+      payload: cleanPrevState ? newStore : { ...bookingInfo, ...newStore }
+    });
+  };
+
+  /**
+   * Send a mutation in order to register the group booking,
+   * and get necessary information to pay for the deposit.
+   */
+  const bookGroup = useMutation<GroupBookingRequestResponse, Error>(async () => {
+    if (!organizerInfo || !bookingInfo?.offers || !bookingInfo?.adultCount) {
+      throw new Error('Something went wrong. Please try selecting your rooms again.');
+    }
+    const { billingInfo, ...restOrganizerInfo } = organizerInfo;
+    const includeBillingInfo = bookingInfo?.invoice;
+
+    console.log(bookingInfo.invoice, billingInfo);
+
+    const result = await bookGroupRequest({
+      organizerInfo: {
+        ...restOrganizerInfo,
+        ...(includeBillingInfo && { billingInfo })
+      },
+      offers: bookingInfo.offers,
+      guestCount: bookingInfo.adultCount,
+      invoice: bookingInfo.invoice ?? false
+    });
+
+    const { depositOptions, serviceId, providerId } = result;
+
+    // TODO: This needs to be revisited for the story to convert to different currencies
+    const quote: Quote = {
+      quoteId: 'dummy',
+      sourceCurrency: 'USD',
+      sourceAmount: depositOptions.usd || 'dummy',
+      targetCurrency: 'dummy',
+      targetAmount: 'dummy',
+      rate: 'dummy'
```

Comment:
possibly missed

Reply:
discussed this with @blecorre and this is what he suggested for now. 

For the price filtering this is also gonna be refacotored on the BE, so then it is likely that we can remove it again.

Reply:
@blecorre let me know if i missed something

Reply:
Well, just to be clear, I did not suggest you to write 'dummy' in it :) But indeed, sending the quote to the front-end adds unnecessary complexity, the logic source/target should stay in simard-pay. And the filtering functionality will add a second quote, bringing even more complexity. So I decided to abstract this complexity for group booking, and to limit the impact of this choice for now on the frontend, the best way was to workaround the Quote and generate a fake one on front end side. I hope we will have time to do the same in the standard flow when developing the filtering with the preferred currency flow so we can get rid of this quote.